build.sbt

```diff
@@ -514,7 +515,50 @@ lazy val scaladoc = configureAsSubproject(project)
   .settings(
     name := "scala-compiler-doc",
     description := "Scala Documentation Generator",
-    includeFilter in unmanagedResources in Compile := "*.html" | "*.css" | "*.gif" | "*.png" | "*.js" | "*.txt" | "*.svg" | "*.eot" | "*.woff" | "*.ttf"
+    includeFilter in unmanagedResources in Compile := "*.html" | "*.css" | "*.gif" | "*.png" | "*.js" | "*.txt" | "*.svg" | "*.eot" | "*.woff" | "*.ttf",
+    libraryDependencies += jqueryDep,
+    resourceGenerators in Compile += Def.task {
+      import java.security.MessageDigest
+      import java.util.Base64
+      val md = MessageDigest.getInstance("SHA-256")
+      val base64encoder = Base64.getEncoder
+
+      // https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity
+      def calsSubResourceIntegrity(input: String): String = {
+        val messageDigest = md.digest(input.getBytes)
+        val base64encoded = base64encoder.encode(messageDigest)
+        new String(base64encoded, "UTF-8")
+      }
+
+      val resourcesFromJars = List(
+        // resourceName, jarName, sha256-SRI
+        ("jquery.min.js", jqueryDep.name, "CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=")
```

Comment:
Could you move this build code into a file in `project`?  That way it keeps the signal-to-noise ratio in build.sbt good (i.e. you don't have some 30 lines of code to micro-manage some jquery details).  Also, it means it's labelled (the file name) and adding comments doesn't further impact the readability of build.sbt.

I'm thinking: make it an object with a method to that takes this jquery details (so maybe a `List[(String, String, String)]`?); but you see what you think works best.

Thank you for spending time on this!

Reply:
Moved SRI checks to `HtmlFactory.scala` in `scaladoc` project.
SRI check will be peformed while generating scaladoc. 
https://github.com/scala/scala/pull/8179/commits/14f0fe5500f55d4d4af5e11b9feedfac0ef6938b

Reply:
Oh... I meant move the build code into a file in the `project/` directory, just to so it doesn't steal attention away from the other important parts of `build.sbt`.

But is it better to have it `HtmlFactory`? 🤔

Reply:
I think now it is simpler in terms of `build.sbt` does not care about files to be bundled, it is concern of `HtmlFactory`.

BTW, moved webjar-related concerns to `project/ScaladocSettings.scala`.
https://github.com/scala/scala/pull/8179/commits/5e5e942a69424d2b99f6bafa78ae0ad7739ae055